### PR TITLE
Add run details

### PR DESCRIPTION
**Pull request by rinckm**
Add the possiblility to define run details which are passed to the
jenkinsfile-runner.
- jobName will be passed as --job-name
- sequenceNumber will be passed as --build-number
- cause will be passed as --cause

Depends on https://github.com/jenkinsci/jenkinsfile-runner/pull/203

As RunDetails are optional this works also with older versions of
jenkinsfile-runner if no RunDetais are defined.

### CODE REVIEW

**Comment by alxsap** [comment]
Shouldn't we also provide/adjust the examples?

File: docs/examples/pipelinerun_details.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    repoUrl: https://github.com/sap-production/demo-pipelines
+    revision: master
+    relativePath: success/Jenkinsfile
+  runDetails:
+    jobName: myName
+    sequenceNumber: 1
+    cause: "my cause"
+  logging:
+    elasticsearch:
+      runID: 1
```
Comment (anbrsap) [inline docs/examples/pipelinerun_details.yaml]: Use the same run id value as in the other examples (in master).

File: docs/examples/pipelinerun_details.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+kind: PipelineRun
+metadata:
+  generateName: ok-
+spec:
+  jenkinsFile:
+    repoUrl: https://github.com/sap-production/demo-pipelines
+    revision: master
+    relativePath: success/Jenkinsfile
+  runDetails:
+    jobName: myName
```
Comment (anbrsap) [inline docs/examples/pipelinerun_details.yaml]: Better use `myJobName`.

However, for examples/tests I prefer values that repeat the field name and have a number at the end, e.g. `jobName1`.

File: docs/examples/pipelinerun_details.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+apiVersion: steward.sap.com/v1alpha1
+kind: PipelineRun
+metadata:
+  generateName: ok-
```
Comment (anbrsap) [inline docs/examples/pipelinerun_details.yaml]: The name prefix should be identical to the name of this example file. This should be changed in all other example files, too.

File: backend-k8s/steward-system/300_clusterTask_steward-jenkinsfile-runner.yaml [hunk 4]
```diff
@@ -41,6 +41,18 @@ spec:
     - name: RUN_NAMESPACE
       description: >
         The namespace of this pipeline run.
+    - name: JOB_NAME
+      description: >
+        Optional job name to be used in the jenkinsfile runner
```
Comment (anbrsap) [inline backend-k8s/steward-system/300_clusterTask_steward-jenkinsfile-runner.yaml]:
```suggestion
        The name of the job this pipeline run belongs to. It is used as the name of the Jenkins job and therefore must be a valid Jenkins job name.
        If null or empty, `job` will be used.
```

File: backend-k8s/steward-system/300_clusterTask_steward-jenkinsfile-runner.yaml [hunk 5]
```diff
@@ -41,6 +41,18 @@ spec:
...
+      default: ""
+    - name: RUN_NUMBER
+      description: >
+        Optional build number of the run
+      default: "1"
+    - name: RUN_CAUSE
+      description: >
+        Optional cause of the run
```
Comment (anbrsap) [inline backend-k8s/steward-system/300_clusterTask_steward-jenkinsfile-runner.yaml]:
```suggestion
        A textual description of the cause of this pipeline run. Will be set as cause of the Jenkins job.
        If null or empty, no cause information will be available.
```

**Pull request merged**